src/common.rs

```diff
@@ -981,9 +982,87 @@ pub fn get_current_pid() -> Result<Pid, &'static str> {
     inner()
 }
 
+/// MAC address for network interface
+#[derive(PartialEq, Eq, Copy, Clone)]
+pub struct MacAddr {
+    data: [u8; 6],
+}
+
+impl MacAddr {
+    pub const UNSPECIFIED: Self = MacAddr::new();
```

Comment:
Ignorant question here: wouldn't it be better to have `Option::None` instead of `UNSPECIFIED`? Is it really relevant to be able to have a `MacAddr` only composed of `0`? If not, then the `new` method would take a `[u8; 6]` argument and check if it's not only composed of `0`, and would return `None` if it's the case.

Reply:
The returned linked list of c `getifaddrs()` does contain mac address with zeros, though `ifconfig` never reveal `00:00:00:00:00:00`...